packages/components/src/internal/components/forms/input/useDataChangeCommentsRequired.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from 'react';
+import { useServerContext } from '../../base/ServerContext';
+import { isProductProjectsEnabled, resolveModuleContext } from '../../../app/utils';
+import { useAppContext } from '../../../AppContext';
+
+export const useDataChangeCommentsRequired = (): { requiresUserComment: boolean } => {
+    const { container, moduleContext } = useServerContext();
+    const { api } = useAppContext();
+    const [requiresUserComment, setRequiresUserComment] = useState<boolean>();
```

Comment:
I'd recommend initializing this to `false` unless `undefined` is an expected part of the contract.

Reply:
If this value has not been set one way or the other, it certainly can be undefined, so initializing it either way doesn't seem to make a huge difference. We also aren't making a strict check for a `true` or `false` value anywhere. But I'm fine with an explicit initialization.